test/y2storage/proposal/autoinst_devices_planner_test.rb

```diff
@@ -98,6 +98,42 @@
           expect(root.reuse).to eq("/dev/sda3")
         end
       end
+
+      context "when the partition to reuse does not exist" do
+        let(:root_spec) do
+          { "create" => false, "mount" => "/", "filesystem" => :ext4, "partition_nr" => 99 }
+        end
+
+        it "adds a new partition" do
+          devices = planner.planned_devices(drives_map)
+          root = devices.find { |d| d.mount_point == "/" }
+          expect(root.reuse).to be_nil
+        end
+
+        it "registers an issue" do
+          planner.planned_devices(drives_map)
+          issue = issues_list.find { |i| i.is_a?(Y2Storage::AutoinstIssues::MissingReusableDevice) }
```

Comment:
I would check that the issues list is empty or does not contain that issue before planning the devices.